fluentd/Dockerfile.centos7

```diff
@@ -59,7 +62,9 @@ ADD parser_viaq_docker_audit.rb viaq_docker_audit.rb /etc/fluent/plugin/
 ADD run.sh generate_syslog_config.rb ${HOME}/
 ADD lib/generate_throttle_configs/lib/*.rb ${HOME}/
 ADD lib/filter_parse_json_field/lib/*.rb /etc/fluent/plugin/
-COPY utils/** ${HOME}/utils
+COPY patch/logger/log.rb ${GEM_HOME}/gems/fluentd-0.12.43/lib/fluent
+COPY patch/logger/supervisor.rb ${GEM_HOME}/gems/fluentd-0.12.43/lib/fluent
+COPY utils/** ${HOME}/utils/
```

Comment:
First, I suggest you do a local build using docker:
```
docker build -t openshift/origin-logging-fluentd:latest -f fluentd/Dockerfile.centos7 fluentd/
```
Then "log in" to the image you just built:
```
docker run -it openshift/origin-logging-fluentd:latest bash
```
then 
```
ls -alrtF /opt/app-root/src/gems/fluentd-0.12.43/lib/fluent
```
What do you see there?

Reply:
Assuming that you mean building the `Dockerfile.centos7` from my fork with all the changes applied. The output of the `ls` command in the running container is:
```
total 236
-rw-r--r--. 1 root root   630 Sep  4 10:43 version.rb
-rw-r--r--. 1 root root  3950 Sep  4 10:43 timezone.rb
-rw-r--r--. 1 root root    36 Sep  4 10:43 time.rb
-rw-r--r--. 1 root root   910 Sep  4 10:43 test.rb
drwxr-xr-x. 2 root root   151 Sep  4 10:43 test/
-rw-r--r--. 1 root root  3354 Sep  4 10:43 system_config.rb
-rw-r--r--. 1 root root  1212 Sep  4 10:43 status.rb
-rw-r--r--. 1 root root  2524 Sep  4 10:43 rpc.rb
-rw-r--r--. 1 root root  7582 Sep  4 10:43 root_agent.rb
-rw-r--r--. 1 root root  1944 Sep  4 10:43 registry.rb
-rw-r--r--. 1 root root 12681 Sep  4 10:43 process.rb
-rw-r--r--. 1 root root  4394 Sep  4 10:43 plugin.rb
drwxr-xr-x. 2 root root  4096 Sep  4 10:43 plugin/
-rw-r--r--. 1 root root 23362 Sep  4 10:43 parser.rb
-rw-r--r--. 1 root root 16655 Sep  4 10:43 output.rb
-rw-r--r--. 1 root root  4943 Sep  4 10:43 mixin.rb
-rw-r--r--. 1 root root  3711 Sep  4 10:43 match.rb
-rw-r--r--. 1 root root   748 Sep  4 10:43 load.rb
-rw-r--r--. 1 root root  1008 Sep  4 10:43 label.rb
-rw-r--r--. 1 root root  1179 Sep  4 10:43 input.rb
-rw-r--r--. 1 root root  8658 Sep  4 10:43 formatter.rb
-rw-r--r--. 1 root root  1665 Sep  4 10:43 filter.rb
-rw-r--r--. 1 root root  4691 Sep  4 10:43 event_router.rb
-rw-r--r--. 1 root root  3404 Sep  4 10:43 event.rb
-rw-r--r--. 1 root root  1002 Sep  4 10:43 env.rb
-rw-r--r--. 1 root root  6321 Sep  4 10:43 engine.rb
-rw-r--r--. 1 root root  4420 Sep  4 10:43 configurable.rb
-rw-r--r--. 1 root root  1443 Sep  4 10:43 config.rb
drwxr-xr-x. 2 root root   199 Sep  4 10:43 config/
drwxr-xr-x. 2 root root    82 Sep  4 10:43 command/
-rw-r--r--. 1 root root  8860 Sep  4 10:43 buffer.rb
-rw-r--r--. 1 root root  5130 Sep  4 10:43 agent.rb
drwxr-xr-x. 1 root root    20 Sep  4 10:43 ../
-rw-r--r--. 1 root root 10044 Sep 11 15:44 log.rb
-rw-r--r--. 1 root root 15709 Sep 11 15:45 supervisor.rb
drwxr-xr-x. 1 root root    27 Sep 11 15:46 ./
```

Reply:
Ok.  I suggest you use `ADD` instead of `COPY` in your Dockerfiles changes.  I have noticed that CI uses a different version of docker, or perhaps docker is configured to have a different build behavior in the CI machines, and sometimes `COPY` does strange things in that environment.  Try using `ADD` instead.

Reply:
Even in the `COPY utils/** ${HOME}/utils/` ? BTW I had to add the last `/` since it would not copy the files.

Also why is the "ci/openshift-jenkins/logging" test running twice? (In jenkins it is called journald why is that?)

Reply:
Do not change `COPY utils/** ${HOME}/utils/` as that seems to be working afaik.  I don't know why.

> Also why is the "ci/openshift-jenkins/logging" test running twice?

Where do you see that?  There should only be a `/logging` and a `/json-file` test - each one should be run only once.

> (In jenkins it is called journald why is that?)

Originally there was only one logging CI test, just called `/logging`, which used docker with `--log-driver=journald`.  When we added support for docker with `--log-driver=json-file` we added a new test called `/json-file`.  The test `ci/openshift-jenkins/logging` is really "deploy docker with --log-driver=journald and test logging", and `ci/openshift-jenkins/json-file` is "deploy docker with --log=driver=json-file and test logging".

Reply:
Thanks for the explanation.
About the test running twice. I confused the times and also the number of tests here: https://openshift-gce-devel.appspot.com/pr/openshift_origin-aggregated-logging/1336
seemed that it is testing it twice.